src/pudl/transform/ferc1.py

```diff
@@ -761,56 +761,48 @@ def reconcile_table_calculations(
+    ]
+    # !!! Add dimensions into the calculation components!!!
+    # First determine what dimensions matter in this table:
+    # usually you can rely on params.subtotal_column to get THE ONE dimension in the
+    # table... BUT some tables have more than one dimension so we grab from all of the
+    # the dims in the transformers. AAAND occasionally the factoid_name is in the dims
+    # wild. i know. so we are grabbing all of the non-factoid dimensions that show up
+    # in the data.
+    dims = [d for d in other_dimensions() if d in df.columns and d != xbrl_factoid_name]
+    calc_idx = ["xbrl_factoid"] + dims
+    table_dims = (
+        df.rename(columns={xbrl_factoid_name: "xbrl_factoid"})[calc_idx]
+        .assign(table_name=table_name)
+        .drop_duplicates(keep="first")
     )
-    if inter_table_calculated_values:
-        logger.warning(
-            "Skipping calculated values because they are inter-table calculations: "
-            f"{inter_table_calculated_values}"
+    if dims:
+        intra_tbl_calcs = make_calculation_dimensions_explicit(
+            intra_tbl_calcs,
+            table_dimensions_ferc1=table_dims,
+            dimensions=dims,
         )
-    intra_tbl_calcs = tbl_meta[
-        tbl_meta.intra_table_calc_flag & (tbl_meta.row_type_xbrl == "calculated_value")
-    ]
     pks = pudl.metadata.classes.Resource.from_id(table_name).schema.primary_key
-    pks_wo_factoid = [col for col in pks if col != xbrl_factoid_name]
-    calculated_dfs = []
-    for calculated_factoid, calculation in zip(
-        intra_tbl_calcs.xbrl_factoid, intra_tbl_calcs.calculations
-    ):
-        calc_df = (
-            pd.merge(
-                df,
-                pd.DataFrame(json.loads(calculation)),
-                left_on=xbrl_factoid_name,
-                right_on="name",
-            )
-            # apply the weight from the calc to convey the sign before summing.
-            .assign(calculated_amount=lambda x: x[params.column_to_check] * x.weight)
-            .groupby(pks_wo_factoid, as_index=False)["calculated_amount"]
-            .sum(min_count=1)
-            .assign(**{xbrl_factoid_name: calculated_factoid})
-        )
-        calculated_dfs.append(calc_df)
-
-    calculated_df = pd.merge(
-        df, pd.concat(calculated_dfs), on=pks, how="left", validate="1:1"
-    )
-    # Force column_to_check to be a float to prevent any hijinks with calculating differences.
-    calculated_df[params.column_to_check] = calculated_df[
-        params.column_to_check
-    ].astype(float)
+    calculated_df = calculate_values_from_components(
+        data=df.rename(columns={xbrl_factoid_name: "xbrl_factoid"}),
+        calculation_components=intra_tbl_calcs,
+        validate="one_to_many",
+        calc_idx=calc_idx,
+        data_idx=[p for p in pks if p != xbrl_factoid_name] + ["xbrl_factoid"],
+        value_col=params.column_to_check,
+    ).rename(columns={"xbrl_factoid": xbrl_factoid_name})
```

Comment:
i tried my darnedest to standardize the process for adding the `calculated_amount` column in this other function. i'm not sure if this is actually simplifying or if it just obscures what is happening in here. i haven't plugged it in over in output land yet but i think it will work well over there as well.